sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -307,7 +307,11 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
 
   private val legacyCastToStr = SQLConf.get.getConf(SQLConf.LEGACY_COMPLEX_TYPES_TO_STRING)
   // The brackets that are used in casting structs and maps to strings
-  private val (leftBracket, rightBracket) = if (legacyCastToStr) ("[", "]") else ("{", "}")
+  protected def leftBracket = if (legacyCastToStr) "[" else "{"
+  protected def rightBracket = if (legacyCastToStr) "]" else "}"
+  protected def elementSpace = " "
```

Comment:
`arrayElementSpace`

Reply:
BTW does it really matter? In TRANSFORM what happens if we add a space between elements? Hive serde fails to parse it?